connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverter.java

```diff
@@ -785,7 +785,7 @@ private Schema inferSchema(JsonNode jsonValue) {
                     return Schema.FLOAT64_SCHEMA;
                 }
             case ARRAY:
-                SchemaBuilder arrayBuilder = SchemaBuilder.array(inferSchema(jsonValue.elements().next()));
+                SchemaBuilder arrayBuilder = SchemaBuilder.array(jsonValue.elements().hasNext() ? inferSchema(jsonValue.elements().next()) : Schema.OPTIONAL_STRING_SCHEMA);
```

Comment:
It might be better to pull the inference out to different line, since it's possible that the `inferSchema` method returns null, and then handle that null case as `SchemaBuilder.array(...)` does not tolerate a null schema.

Regardless, this is just one of the assumptions that will have to be documented in the KIP. I suspect these assumptions are where most of the contention will be. It's not clear that assuming an empty array is a String array is necessarily the best option.